### PR TITLE
Tower rework

### PR DESCRIPTION
**Pull request by Patgar1**

### CODE REVIEW

File: game/src/main/java/xyz/openatbp/extension/game/actors/UserActor.java [hunk 1]
```diff
@@ -648,21 +731,43 @@ public boolean canUseAbility(int ability){
         return this.canCast[ability-1];
     }
 
-    public boolean isDash(String avatar, int ability){ //all chars except fp
-        switch (avatar){
+    public boolean isDash(String avatar, int ability) { //all chars except fp
+        switch (avatar) {
```
Comment (CakeLancelot) [inline game/src/main/java/xyz/openatbp/extension/game/actors/UserActor.java:735]: Perhaps this should be a different PR, but we should have a function that takes in the avatar name and strips any skin related text so that cases like these could be simplified.

File: game/src/main/java/xyz/openatbp/extension/game/actors/UserActor.java [hunk 2]
```diff
@@ -910,6 +1018,42 @@ public int getSpellDamage(JsonNode attackData){
...
+        double y = location.getY();
+        double dx = dest.getX() - location.getX();
+        double dy = dest.getY() - location.getY();
+        double length = Math.sqrt(dx * dx + dy * dy);
+        double unitX = dx/length;
+        double unitY = dy/length;
+        double extendedX = x + abilityRange * unitX;
+        double extendedY = y + abilityRange * unitY;
+        Point2D lineEndPoint = new Point2D.Double(extendedX,extendedY);
+        //double distance = location.distance(lineEndPoint);
```
Comment (CakeLancelot) [inline game/src/main/java/xyz/openatbp/extension/game/actors/UserActor.java:1032]: Go ahead and remove this if it's unused

File: game/src/main/java/xyz/openatbp/extension/game/actors/UserActor.java [hunk 3]
```diff
@@ -910,6 +1018,42 @@ public int getSpellDamage(JsonNode attackData){
...
```
Comment (CakeLancelot) [inline game/src/main/java/xyz/openatbp/extension/game/actors/UserActor.java:1050]: Same as above

File: game/src/main/java/xyz/openatbp/extension/game/champions/CinnamonBun.java [hunk 4]
```diff
@@ -126,29 +126,15 @@ public void useAbility(int ability, JsonNode spellData, int cooldown, int gCoold
...
+                double slideY = Champion.getAbilityLine(this.location,dest,1.5f).getP2().getY();
                 float rotation = getRotation(dest);
-                Point2D directionVector = new Point2D.Double(endLocation.getX() - startLocation.getX(), endLocation.getY() - startLocation.getY());
-                double length = Math.sqrt(directionVector.getX() * directionVector.getX() + directionVector.getY() * directionVector.getY());
-                double normalizedX = directionVector.getX() / length;
-                double normalizedY = directionVector.getY() / length;
-                float desiredX = (float) (startLocation.getX() + 1.5f * normalizedX);
-                float desiredY = (float) (startLocation.getY() + 1.5f * normalizedY);
-                ExtensionCommands.createActorFX(this.parentExt,this.room,this.id,"fx_target_rect_7",5000,this.id+"w",false,"",true,true,this.team);
-                //ExtensionCommands.createWorldFX(this.parentExt,this.room,this.id,"fx_target_rect_7",this.id+"w",5000,(float)startLocation.getX(),(float)startLocation.getY(),false,this.team,rotation);
```
Comment (CakeLancelot) [inline game/src/main/java/xyz/openatbp/extension/game/champions/CinnamonBun.java:147]: Same as above

File: game/src/main/java/xyz/openatbp/extension/game/actors/UserActor.java (hunk 1 above)
Comment (Patgar1) [reply game/src/main/java/xyz/openatbp/extension/game/actors/UserActor.java:735]: Got it, I'll include this in the next commit.

File: game/src/main/java/xyz/openatbp/extension/game/champions/RattleBalls.java [hunk 5]
```diff
@@ -263,7 +265,9 @@ protected void spellQ() {
                 SmartFoxServer.getInstance().getTaskScheduler().schedule(flipDelay,qTime,TimeUnit.MILLISECONDS);
             }else{
                 abilityEnded();
-                canCast[0] = true;
+                canCast[0] = false;
```
Comment (Darklit) [inline game/src/main/java/xyz/openatbp/extension/game/champions/RattleBalls.java:268]: This would make it so Rattle can't use any abilities after his q ends. Is this the intended goal?
